rfd/0147-repo-fork-syncing.md

```diff
@@ -0,0 +1,60 @@
+---
+authors: Mike Jensen (mike.jensen@goteleport.com)
+state: draft
+---
+ 
+# RFD 144 - Automatic Repo Syncing
+ 
+## What
+ 
+This RFD describes how we can sync a repo continuously and in an automatic way.  This can be used to provide private copies of a repo.
+ 
+## Why
+ 
+We currently face a few frictions which could be addressed through a more robust repo sync mechanism:
+1. Our internal `teleport-private` fork used for security development must be manually updated
+2. Some tooling (Dependabot) does not allow scanning for non-default branches, necessitating manual efforts to manage our release branches
+3. Because much of the CI and scanning configuration is committed into the repo, we face noise from duplicate alerts when we use a repo for testing or as an internal fork
+ 
+## Details
+ 
+As such we need a solution that can provide the following:
+* Automatic and regular updates to a repo to ensure it's in sync
+* The ability to apply changes to the upstream repo (for example disable Actions that we don't want running on the repo copy, or enable additional scanning / actions)
+ 
+### Branch Structure
+ 
+* `main` - This will be the repo's default branch.  This branch will contain the upstream history in addition to commits needed for the sync process or other custom changes on the repo copy
```

Comment:
Why are we using `main` here when `master` is the default branch for `teleport`?  I feel like mixing the two paradigms is going to make things more confusing.

Reply:
This is configurable.  The reason for the difference is `teleport` uses `master`, but new repos default to `main`.  I have no concerns if we want to override that default in the new repos and continue to use `master`.

Reply:
I updated to `master` so that it's consistent, but indicated that this can be configured.  If we choose to use `teleport` that's fine as well.

Reply:
I would go with `teleport` to avoid confusion, as it's been already used for some of our other forks and is an unlikely branch name in the upstreams.

Reply:
Are you suggesting that the default branch of `teleport-private` becomes `teleport`?

I don't really have a preference, but was trying to minimize changes if they are not necessary.

Reply:
There's a distinction between the "automatic repo mirroring" proposal, which I understand to be a more generic mechanism, and whatever we do for teleport-private. Most of my comments are thinking of the former, not the latter.

Reply:
I prefer to keep it as written, I think highlighting it's configurable and referencing the `teleport-private` case as the example probably makes this easiest to read.

Reply:
Sounds good for security-scanning, "private fix but never land PRs" kind of repos.